lua/mapvote/client/vgui/searchable_scrollpanel.lua

```diff
@@ -0,0 +1,47 @@
+---@class SearchableScrollPanel : DScrollPanel
+local PANEL = {}
+function PANEL:Init()
+    self.items = {}
+    self.filter = ""
+end
+
+function PANEL:SetSearch( filter )
+    self.filter = filter
+end
+
+---@param entry Panel
+function PANEL:BindToEntry( entry )
+    entry.OnValueChanged = function( _, value )
+        timer.Create( "MapVote_SearchableScrollPanel_Refresh", 0.5, 1, function()
+            self:SetSearch( value )
+            self:Refresh()
+        end ) -- this means that the search will only refresh after 2 seconds of no typing
```

Comment:
Is it meant to be a 2s delay?
```suggestion
        timer.Create( "MapVote_SearchableScrollPanel_Refresh", 2, 1, function()
            self:SetSearch( value )
            self:Refresh()
        end ) -- this means that the search will only refresh after 2 seconds of no typing
```

Reply:
yeah its just an arbitrary delay so we dont bother recreating the scroll panel while they arent finished typing

Reply:
replaced  with descriptive local variable instead of comment